opam does not respect `--no-checksums`

OPAM does not seem to respect the `--no-checksums` flag (or the $OPAMNOCHECKSUMS env variable).

I don't have enough knowledge to produce a minimal test case, but at this point in time the following triggers the incorrect behaviour for me:

```
opam install --no-checksums coq-stdpp.1.1.0 -vv
+ /home/dan/.opam/4.07.0/bin/ocamlc "-vnum"
- 4.07.0
The following actions will be performed:
  ∗ install   ocaml-config 1
  ↻ recompile ocaml        4.07.0
  ∗ install   ocamlfind    1.8.0             [required by coq]
  ∗ install   camlp5       7.06.10-g84ce6cc4 [required by coq]
  ∗ install   num          1.1               [required by coq]
  ∗ install   coq          8.8.2             [required by coq-stdpp]
  ∗ install   coq-stdpp    1.1.0
===== ∗ 6   ↻ 1 =====
Do you want to continue? [Y/n] y

<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[camlp5.7.06.10-g84ce6cc4] found in cache
[coq.8.8.2] found in cache
+ /bin/tar "xfz" "/home/dan/.opam/download-cache/sha256/3a/3a9b90684a710b23b2f83f020304f0fda48219725d7a9ea881ae49effd9afdeb" "-C" "/tmp/opam-22016-ccabea"
[num.1.1] found in cache
+ /bin/tar "xfz" "/home/dan/.opam/download-cache/md5/71/710cbe18b144955687a03ebab439ff2b" "-C" "/tmp/opam-22016-c61e7b"
[ocamlfind.1.8.0] found in cache
+ /bin/tar "xfz" "/home/dan/.opam/download-cache/md5/5d/5d693cd1953a0dd74920b43d183bc26c" "-C" "/tmp/opam-22016-7afd23"
+ /bin/tar "xfz" "/home/dan/.opam/download-cache/md5/a7/a710c559667672077a93d34eb6a42e5b" "-C" "/tmp/opam-22016-a00279"
+ /usr/bin/curl "--write-out" "%{http_code}\\n" "--retry" "3" "--retry-delay" "2" "--user-agent" "opam/2.0.0" "-L" "-o" "/home/dan/.opam/iris31/.opam-switch/sources/coq-stdpp.1.1.0/archive.tar.gz.part" "https://gitlab.mpi-sws.org/robbertkrebbers/coq-stdpp/repository/coq-stdpp-1.1.0/archive.tar.gz"
-   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
-                                  Dload  Upload   Total   Spent    Left  Speed
100   152  100   152    0     0    548      0 --:--:-- --:--:-- --:--:--   546
100  151k  100  151k    0     0   375k      0 --:--:-- --:--:-- --:--:--  375k
- 200
[ERROR] coq-stdpp.1.1.0: Checksum mismatch for
        https://gitlab.mpi-sws.org/robbertkrebbers/coq-stdpp/repository/coq-stdpp-1.1.0/archive.tar.gz:
          expected md5=2959ffffb9ca8814cf9a388436a28867
          got      md5=2a24643f073f0975597dfbe6cbbc63e5
[ERROR] The sources of the following couldn't be obtained, aborting:
          - coq-stdpp.1.1.0:
            Checksum mismatch

'opam install --no-checksums coq-stdpp.1.1.0 -vv' failed.
```

Expected behaviour: opam does not check for the checksum mismatch.

`opam config report` output:
```
# opam config report
# opam-version      2.0.0 
# self-upgrade      no
# system            arch=x86_64 os=linux os-distribution=ubuntu os-version=18.04
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              4
# repositories      2 (http), 1 (version-controlled) (default repo at 7ed1cc7a)
# pinned            1 (git), 1 (rsync)
# current-switch    4.07.0
```

--- a/src/repository/opamRepository.ml
+++ b/src/repository/opamRepository.ml
@@ -192,7 +192,8 @@ let pull_from_upstream
   )
   @@| function
   | (Result (Some file) | Up_to_date (Some file)) as ret ->
-    if validate_and_add_to_cache label url cache_dir file checksums then
+    if OpamRepositoryConfig.(!r.force_checksums) = Some false
+    || validate_and_add_to_cache label url cache_dir file checksums then
       (OpamConsole.msg "[%s] %s from %s\n"
          (OpamConsole.colorise `green label)
          (match ret with Up_to_date _ -> "no changes" | _ -> "downloaded")

